wordpress-mobile/WordPress-Android
# Show Jetpack stats module is disabled view

Fixes #14942

This PR adds support for communicating that the Stats module is disabled instead of showing a generic error after the user taps on "Stats" from the My Site view. It also supports enabling the module. The accompanying FluxC PR: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2062

Jetpack not installed error | Stats not enabled error
-- | --
<img width="250" height="500" alt="light" src="https://user-images.githubusercontent.com/506707/125825017-6ff2211e-80e6-41e8-8e1d-501916a8d176.png"> | <img width="250" height="500" alt="dark" src="https://user-images.githubusercontent.com/506707/125825429-d2305691-2a42-4bfa-951b-5e0d27f194ea.png">

**Noteworthy**
- `StatsModuleActionableView` is copied and slimmed down version of `EmptyActionableView` that includes a progress bar. I thought about introducing the ProgressBar to EmptyActionableView, but that sort of change was high risk . New attributes and a layout file were also added, but it uses the aev styles for consistency.
- Calls to FluxC are made via `StatsModuleActivateUseCase` and the response is communicated as `StatsModuleActivateRequestState`
- `StatsViewModel` exposes a new observer `statsModuleUiModel` to communicate with `StatsFragment` for the setting view visibility.   

To test:
- Disable the stats module on your website (see: https://jetpack.com/support/control-jetpack-features-on-one-page/)
- Open the Jetpack app and go to My Site > Stats
- Notice the "Looking for Stats?" view is displayed

**Merge Instructions**
- Review corresponding FluxC PR
- Merge the FluxC PR to develop, create a tag, update it in the WPAndroid build.gradle . I can do this part after the PR is approved - unless said reviewer feels up to it. :)
- Remove Not Ready for Merge label
- Merge this PR

## Regression Notes
1. Potential unintended areas of impact

- Stats view is not shown

2. What I did to test those areas of impact (or what existing automated tests I relied on)

- Manually tested
- Introduced automated tests: StatsViewModelTest, StatsModuleActivateUseCaseTest

3. What automated tests I added (or what prevented me from doing so)
See above.

PR submission checklist:
